lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -191,6 +165,52 @@ def self.is_firebasecmd_supported?(cmd)
 
         true
       end
+
+      def self.get_token
+        client = Signet::OAuth2::Client.new(
+          token_credential_uri: 'https://oauth2.googleapis.com/token',
+          client_id: CLIENT_ID,
+          client_secret: CLIENT_SECRET,
+          refresh_token: ENV["FIREBASE_TOKEN"]
+        )
+        client.fetch_access_token!
+        return client.access_token
+      rescue Signet::AuthorizationError => error
+        UI.crash!("Failed to retrieve FIREBASE_TOKEN")
+      end
+
+      def self.get_app(params)
+        token = get_token
+
+        # begin
+        base_url = "https://firebaseappdistribution.googleapis.com"
+        subD = "/v1alpha/apps/"
+        appId = (params[:app]).to_s
+        url = subD + appId
+
+        connection = Faraday.new(url: base_url) do |conn|
+          conn.response(:json, parser_options: { symbolize_names: true })
+          conn.response(:raise_error)
+          conn.adapter(Faraday.default_adapter)
+        end
+
+        response = connection.get(url) do |request|
+          request.headers["Authorization"] = "Bearer " + token
+        end
+
+        contactEmail = response.body[:contactEmail]
+        UI.message(contactEmail)
+
+        if contactEmail.strip.empty?
+          UI.error("Empty contact email")
```

Comment:
Thinking ahead to how this method will be used: should there be a return value from this method? What would it be in this case, where the app is invalid because there's no contact email? Or maybe the return value of this method is the app itself, and this check happens in the main body of the action.